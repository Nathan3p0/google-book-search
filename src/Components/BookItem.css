.book_item {
    border-bottom: 1px solid #cccccc;
}

.book_item h2 {
    text-align: center;
}

.book_details {
    display: flex;
    justify-content: flex-start;
    align-items: flex-start;
    text-align: center;
}

.book_details > * {
    margin: 0.5rem;
}

.book_details p:first-child {
    margin-top: 0;
    padding-top: 0;
}

.book_info {
    width: 100%;
    max-width: 500px;
}

.book_info p:last-of-type {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
}