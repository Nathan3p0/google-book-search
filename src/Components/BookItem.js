import React from 'react';
import './BookItem.css'

const BookItem = (props) => {
    return ( 
        <li className="book_item">
            <h2>{props.bookInfo.title}</h2>
            <div className="book_details">
                <img src={props.bookInfo.thumbnail} alt='Book cover' />
                <div className="book_info">
                    <p>Author: {props.bookInfo.authors}</p>
                    <p>Price: {props.bookInfo.price}</p>
                    <p>{props.bookInfo.snippet}</p>
                </div>
            </div>
        </li>
     );
}

BookItem.defaultProps = {
    bookInfo : {
        title: '',
        thumbnail: '',
        authors: '',
        price: '',
        snippet: ''
    }
}
 
export default BookItem;