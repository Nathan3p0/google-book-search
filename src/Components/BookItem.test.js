import React from 'react'
import ReactDOM from 'react-dom'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import BookItem from './BookItem';

describe(`Book Item Component`, () => {
    const booksProp = {
        title: 'Llamas',
        thumbnail: 'http://books.google.com/books/content?id=3pJ-YdPLI2UC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api',
        authors: 'Peru',
        price: '$15,000',
        snippet: 'Llamas can wear pajamas'
    }
  
    it('renders without errors', () => {
      const div = document.createElement('div');
      ReactDOM.render(<BookItem />, div);
      ReactDOM.unmountComponentAtNode(div);
    })
  
    it('renders empty given no book detail', () => {
      const wrapper = shallow(<BookItem />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  
    it('renders with book info properly', () => {
      const wrapper = shallow(<BookItem bookInfo={booksProp} />)
      expect(toJson(wrapper)).toMatchSnapshot()
    })
  })