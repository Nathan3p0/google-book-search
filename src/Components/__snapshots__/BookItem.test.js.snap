// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book Item Component renders empty given no book detail 1`] = `
<li
  className="book_item"
>
  <h2 />
  <div
    className="book_details"
  >
    <img
      alt="Book cover"
      src=""
    />
    <div
      className="book_info"
    >
      <p>
        Author: 
      </p>
      <p>
        Price: 
      </p>
      <pre />
    </div>
  </div>
</li>
`;

exports[`Book Item Component renders with book info properly 1`] = `
<li
  className="book_item"
>
  <h2>
    Llamas
  </h2>
  <div
    className="book_details"
  >
    <img
      alt="Book cover"
      src="http://books.google.com/books/content?id=3pJ-YdPLI2UC&printsec=frontcover&img=1&zoom=5&edge=curl&source=gbs_api"
    />
    <div
      className="book_info"
    >
      <p>
        Author: 
        Peru
      </p>
      <p>
        Price: 
        $15,000
      </p>
      <pre>
        Llamas can wear pajamas
      </pre>
    </div>
  </div>
</li>
`;
